use extras_require

chie8842 opened the pull request:
It's only a propal but how about using `extras_require` to separate package dependencies.
See.
https://qiita.com/ledmonster/items/74886ac2f51fb2e5fd64

himkt commented:
Thanks for your suggestion. 👍 
Personally, I think tiny_tokenizer should install word tokenizer libraries
such as `natto-py`, `kytea`, `sentencepiece` and `sudachi-py` by default.
(Because I want to be ready to use all tokenizers by executing `pip install tiny_tokenizer`)

This is why I offer an option to ignore installing them. (not an option to install them.)
What do you think? @chie8842

chie8842 commented:
I'd like to use tiny_tokenizer for my application, but I'd like to keep my application as small as possible.
small specifies that small dependencies, small container size.

But the other side, it may more convenient for research usage with install all dependencies.
And I can fit to my application with `BUILD_WORD_TOKENIZER`.

himkt commented on setup.py:
```diff
@@ -11,22 +11,19 @@
     description="Tiny Word/Sentence Tokenizer",
     author="himkt",
     author_email="himkt@klis.tsukuba.ac.jp",
     install_requires=install_requires,
```
Please remove this.

himkt replied:
410c1d3

himkt commented:
@chie8842 

> using extras_require is more understandable than including BUILD_WORD_TOKENIZER to application.
After considering, I support your suggestion because I found it to be reasonable.
Then, could you please update `.travis.yml` such that CI could pass?

I think it could be done by:

1. Update env section (https://github.com/himkt/tiny_tokenizer/blob/master/.travis.yml#L16-L18)
```
env:
  - PACKAGE_NAME=tiny_tokenizer
  - PACKAGE_NAME=tiny_tokenizer[mecab,kytea,sudachi-py,sentencepiece]
```

2. Update installation statement (https://github.com/himkt/tiny_tokenizer/blob/master/.travis.yml#L21)
```
pip install $PACKAGE_NAME
```

Thanks!

chie8842 commented:
I added `all` key to install all dependencies.
so we can install all tokenizers with `pip install tiny_tokenizer[all]`.
updated `.travis.yml`, too:)

himkt commented:
Thanks for updating! LGTM.
